src/Microsoft.AspNet.OData.Shared/GetNextPageHelper.cs

```diff
@@ -29,7 +29,7 @@ internal static Uri GetNextPageLink(Uri requestUri, IEnumerable<KeyValuePair<str
 
             foreach (KeyValuePair<string, string> kvp in queryParameters)
             {
-                string key = kvp.Key.ToLowerInvariant();
+                string key = kvp.Key.StartsWith("$")? kvp.Key.ToLowerInvariant() : kvp.Key;
```

Comment:
Is this a reliable way to tell whether the query param is not a standard OData query param. I think in 4.01, OData allows "official" query params without the `$` prefix.